### PR TITLE
Add note on preserving API response case style for kinds from non-asset APIs

### PR DESCRIPTION
**Pull request by phalbert**
# Description

- Introduced a new section in the examples documentation to explain the use of `preserveApiResponseCaseStyle` in the selector configuration.
- Provided examples for `pubsub.googleapis.com/Subscription` and `pubsub.googleapis.com/Topic`.
- Clarified the purpose of this configuration to maintain camelCase format from the GCP API response.

## Updated docs pages

Please also include the path for the updated docs

- GCP Installation - Mapping Extra Resources (`/docs\build-your-software-catalog\sync-data-to-catalog\cloud-providers\gcp\examples\mapping_extra_resources.md`)

### CODE REVIEW

File: docs/build-your-software-catalog/sync-data-to-catalog/cloud-providers/gcp/examples/mapping_extra_resources.md [hunk 1]
```diff
@@ -148,3 +148,32 @@ Create an integration configuration for the resource. The integration configurat
...
+      query: 'true'
+      preserveApiResponseCaseStyle: 'true'
+    port:
+      entity:
+        mappings:
+          identifier: .name
+          title: .name | split("/") | last
+```
+
+This ensures that properties maintain their original camelCase format from the GCP API response.
```
Comment (mk-armah) [inline docs/build-your-software-catalog/sync-data-to-catalog/cloud-providers/gcp/examples/mapping_extra_resources.md:179]: List the APIs affected and their endpoints, Example

- Projects: `cloudresourcemanager.googleapis.com/Project`

**Review by mk-armah — CHANGES_REQUESTED** [review]
Left some comments and recommendations

File: docs/build-your-software-catalog/sync-data-to-catalog/cloud-providers/gcp/examples/mapping_extra_resources.md [hunk 2]
```diff
@@ -148,3 +148,44 @@ Create an integration configuration for the resource. The integration configurat
...
+      query: 'true'
+      preserveApiResponseCaseStyle: 'true'
+    port:
+      entity:
+        mappings:
+          identifier: .name
+          title: .name | split("/") | last
+```
+
+When `preserveApiResponseCaseStyle` is not set or set to `false` (default behavior), all property names will be converted to camelCase format regardless of their original case style in the API response.
```
Comment (mk-armah) [inline docs/build-your-software-catalog/sync-data-to-catalog/cloud-providers/gcp/examples/mapping_extra_resources.md:191]: That's the default behavior ? we are not keeping backward compatibility  ?

File: docs/build-your-software-catalog/sync-data-to-catalog/cloud-providers/gcp/examples/mapping_extra_resources.md [hunk 3]
```diff
@@ -148,3 +148,44 @@ Create an integration configuration for the resource. The integration configurat
               machineType: ".machineType"
         		# highlight-end
         ```
+
+
+
+### Case Style Preservation
+
+Certain APIs return properties in specific case styles (e.g., snake_case). By default, the GCP integration will convert these properties to camelCase. However, you can preserve the original API response case style for the following APIs:
+
```
Comment (mk-armah) [inline docs/build-your-software-catalog/sync-data-to-catalog/cloud-providers/gcp/examples/mapping_extra_resources.md:157]: It converts by default ?

**Review by mk-armah — COMMENTED** [review]
need some clarifications

**Review by mk-armah — APPROVED** [review]
LGTM

File: docs/build-your-software-catalog/sync-data-to-catalog/cloud-providers/gcp/examples/mapping_extra_resources.md [hunk 4]
```diff
@@ -148,3 +148,46 @@ Create an integration configuration for the resource. The integration configurat
               machineType: ".machineType"
         		# highlight-end
         ```
+
+
+
+### Case Style Preservation
+
+For resources listed below, the GCP Integration provides flexibility to either transform the properties to snake_case or preserve the original case style (e.g., camelCase) returned by the API for consistency with other resources.
```
Comment (hadar-co) [inline docs/build-your-software-catalog/sync-data-to-catalog/cloud-providers/gcp/examples/mapping_extra_resources.md:156]:
```suggestion
For the resources listed below, this integration provides flexibility to either transform the properties to `snake_case` or preserve the original case style (e.g., `camelCase`) returned by the API for consistency with other resources.
```

File: docs/build-your-software-catalog/sync-data-to-catalog/cloud-providers/gcp/examples/mapping_extra_resources.md [hunk 5]
```diff
@@ -148,3 +148,46 @@ Create an integration configuration for the resource. The integration configurat
...
+
+This feature applies to the following kinds:
```
Comment (hadar-co) [inline docs/build-your-software-catalog/sync-data-to-catalog/cloud-providers/gcp/examples/mapping_extra_resources.md:158]:
```suggestion
This applies to the following kinds:
```

**Review by hadar-co — APPROVED** [review]

**Pull request merged**